src/dhtproto/node/neo/request/Mirror.d

```diff
@@ -354,40 +361,56 @@ public abstract scope class MirrorProtocol_v0
 
     /***************************************************************************
 
-        Request handler. Reads the initial message from the client, responds to
-        the client with a status code, and starts the request handling fibers.
+        Called by the connection handler immediately after the request code and
+        version have been parsed from a message received over the connection.
+        Allows the request handler to process the remainder of the incoming
+        message, before the connection handler sends the supported code back to
+        the client.
+
+        Note: the initial payload is a slice of the connection's read buffer.
+        This means that when the request-on-conn fiber suspends, the contents of
+        the buffer (hence the slice) may change. It is thus *absolutely
+        essential* that this method does not suspend the fiber. (This precludes
+        all I/O operations on the connection.)
 
         Params:
-            connection = connection to client
-            msg_payload = initial message read from client to begin the request
-                (the request code and version are assumed to be extracted)
+            init_payload = initial message payload read from the connection
 
     ***************************************************************************/
 
-    final public void handle ( RequestOnConn connection,
-        Const!(void)[] msg_payload )
+    public void preSupportedCodeSent ( Const!(void)[] init_payload )
     {
-        this.connection = connection;
-
         // Parse initial message from client.
-        bool start_suspended;
         cstring channel;
-        this.connection.event_dispatcher.message_parser.parseBody(msg_payload,
-            start_suspended, channel, this.initial_refresh,
+        this.connection.event_dispatcher.message_parser.parseBody(init_payload,
+            this.start_suspended, channel, this.initial_refresh,
             this.periodic_refresh_s);
 
-        auto ok = this.prepareChannel(channel);
+        this.initialised_ok = this.prepareChannel(channel);
+    }
+
+    /***************************************************************************
+
+        Called by the connection handler after the supported code has been sent
+        back to the client.
 
+    ***************************************************************************/
+
+    private bool initialised_ok;
+    private bool start_suspended;
```

Comment:
Order of the (undocumented) fields and the method docs doesn't match.

Reply:
Oh yeah, they should be documented.

Reply:
Updated.